src/View/Router.ts

```diff
@@ -34,6 +34,7 @@ export class Router {
       Exchange.BatchReaderReadNextDirectorySuccess,
       this.onBatchReaderReadNextDirectorySuccess
     )
+    await this.messaging.consume(Exchange.BatchWriterCreateNextBatchSuccess, this.onBatchWriterCreateNextBatchSuccess)
```

Comment:
@kennylavender we were missing adding the ipfsDirectoryHash to works when the Claim Batch is being created in this node rather than being downloaded from others

Reply:
I did that on purpose because after a transaction is created and added to a block, the block would then be read and we would get the ipfsDirectoryHash that way. I thought about doing both but wasn't sure if it was redundant. I think its fine to have both ways so Il merge this too.

Reply:
Yep, we want to have both just so we don't have to wait 10 minutes to see that data in the API.

In the future we'll be able to add block confirmations to ensure the data is consistent. For now, if there are discrepancies between the data read from a block and the one written on the creation process we can either log it or create a ticket about it so we don't forget.